pkg/plugin/plugin.go

```diff
@@ -78,6 +82,55 @@ type Info struct {
 	ConfigAttributes json.RawMessage `db:"config_attrs" json:"config_attrs"` // ConfigOption(s) as json-encoded list
 }
 
+// NewInfo constructs a new Info to be used in Plugin.GetInfo.
+//
+// The configOptions will be enriched with defaults from the typePtr, being a pointer to the Plugin implementation.
+func NewInfo(typePtr interface{}, name, version, author string, configOptions []*ConfigOption) (*Info, error) {
+	err := defaults.Set(typePtr)
+	if err != nil {
+		return nil, err
+	}
+
+	typeDefaults := make(map[string]any)
+	typeRefT := reflect.TypeOf(typePtr).Elem()
+	typeRefV := reflect.ValueOf(typePtr)
+	for i := 0; i < typeRefT.NumField(); i++ {
+		if !typeRefT.Field(i).IsExported() {
+			continue
+		}
+
+		jsonVal, jsonOk := typeRefT.Field(i).Tag.Lookup("json")
+		_, defaultOk := typeRefT.Field(i).Tag.Lookup("default")
+
+		if !jsonOk {
+			return nil, fmt.Errorf("exported field %q misses json struct tag, set to \"-\" to ignore", typeRefT.Field(i).Name)
+		}
+		if jsonVal == "-" || !defaultOk {
+			continue
+		}
+
+		typeDefaults[jsonVal] = typeRefV.Elem().Field(i).Interface()
```

Comment:
I would just use the value from `typeRefT.Field(i).Tag.Lookup(‘default’)` instead of this additional reflection, and you wouldn't have to call `defaults.Set()` at the beginning of the function as well.

Reply:
In theory, we already do so. However, currently there are no default - or any - values which are not strings.

https://github.com/Icinga/icinga-notifications/blob/7ff9196bda971ea6741eb42b000c8f7fc4bf1917/pkg/plugin/plugin.go#L31-L34